lib/Dialect/FIRRTL/Transforms/IMDeadCodeElim.cpp

```diff
@@ -399,6 +417,41 @@ void IMDeadCodeElimPass::rewriteModuleSignature(FModuleOp module) {
   numRemovedPorts += deadPortIndexes.size();
 }
 
+void IMDeadCodeElimPass::eraseEmptyModule(FModuleOp module) {
+  // Public modules cannot be erased.
+  if (module.isPublic())
+    return;
+
+  // If the module doesn't have arguments, operations or annotations, we
+  // consider it to be dead.
+  if (liveModules.count(module) || !module.annotations().empty())
+    return;
```

Comment:
1. In the comment, you don't check for operations or arguments any more 
2. Can we check`!module.annotations().empty()` in `populateModuleBody` and add the module to the `liveModules` set if it has annotations?